File: awsiot-test/sub.py
# -*- coding:utf8 -*-
from AWSIoTPythonSDK.MQTTLib import AWSIoTMQTTClient
import os
import time

def customCallback(client, userdata, message):
    print('message: ')
    print(message.payload)
    print('topic: ')
    print(message.topic)
    print('--------------\n\n')

myMQTTClient = AWSIoTMQTTClient('python')
myMQTTClient.configureEndpoint(os.environ['AWSIOT_HOST'], 8883)
myMQTTClient.configureCredentials('keys/G5.pem', 'keys/private.pem.key', 'keys/certificate.pem.crt')
myMQTTClient.configureOfflinePublishQueueing(-1)
myMQTTClient.configureDrainingFrequency(2)
myMQTTClient.configureConnectDisconnectTimeout(10)
myMQTTClient.configureMQTTOperationTimeout(5)
myMQTTClient.connect()
while True:
    myMQTTClient.subscribe("test", 1, customCallback)
    time.sleep(1)

File: awsiot-test/sub.js
var awsIot = require('aws-iot-device-sdk');

var device = awsIot.device({
   //keyPath: 'keys/private.pem.key',
  //certPath: 'keys/certificate.pem.crt',
    //caPath: 'G5.pem',
  clientId: 'nouser' + (Math.floor((Math.random() * 100000) + 1)),
      host: process.env.AWSIOT_HOST,
  protocol: 'wss',
  accessKeyId: process.env.AWSIOT_ACCESSKEYID,
  secretKey: process.env.AWSIOT_SECRETKEY
});

device.on('connect', function() {
  console.log('connect');
  device.subscribe('test');
});

device.on('message', function(topic, payload) {
  console.log(payload.toString());
});
